test/extensions/filters/http/transformation/inja_transformer_test.cc

```diff
@@ -565,6 +601,41 @@ TEST(InjaTransformer, ParseBodyListUsingContext) {
   EXPECT_EQ(body.toString(), "321");
 }
 
+TEST(InjaTransformer, ParseFromClusterMetadata) {
+  Http::TestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+  TransformationTemplate transformation;
+  transformation.mutable_body()->set_text("{{clusterMetadata(\"key\")}}");
+
+  InjaTransformer transformer(transformation);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  
+  envoy::config::core::v3::Metadata meta;
+  meta.mutable_filter_metadata()->insert({SoloHttpFilterNames::get().Transformation, MessageUtil::keyValueStruct("key", "val")});
+  ON_CALL(*callbacks.cluster_info_, metadata()).WillByDefault(testing::ReturnRefOfCopy(meta));
+
+  Buffer::OwnedImpl body("1");
+  transformer.transform(headers, body, callbacks);
+  EXPECT_EQ(body.toString(), "val");
+}
+
+TEST(InjaTransformer, ParseFromNilClusterMetadata) {
+  Http::TestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+  TransformationTemplate transformation;
+  transformation.mutable_body()->set_text("{{clusterMetadata(\"key\")}}");
+
+  InjaTransformer transformer(transformation);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  envoy::config::core::v3::Metadata* meta{};
+
+  ON_CALL(*callbacks.cluster_info_, metadata()).WillByDefault(testing::ReturnRef(*meta));
```

Comment:
*meta is derefing a nil pointer which is techinically undefined behavior.
can we change to?
```
WillByDefault(testing::Return(nullptr));
```

Reply:
this won't compile, will update with similar